src/process.rs

```diff
@@ -409,11 +413,33 @@ where
 {
     /// Shut down process
     pub fn shutdown(&self) {
+        self.shutdown_timeout(Duration::new(0, 0))
+            .expect("no timeout specified")
+    }
+
+    /// Shut down process with a timeout
+    pub fn shutdown_timeout(&self, duration: Duration) -> Result<(), ReceiveError> {
         // Create new message buffer.
-        unsafe { host::api::message::create_data(Tag::none().id(), 0) };
-        Bincode::encode(&Sendable::Shutdown).unwrap();
-        // Send the message
-        unsafe { host::api::message::send(self.process.id()) };
+        let tag = Tag::new();
+        unsafe { host::api::message::create_data(tag.id(), 0) };
+
+        // Create reference to self
+        let this: Process<()> = unsafe { Process::from_id(host::api::process::this()) };
+
+        Bincode::encode(&Sendable::Shutdown(this)).unwrap();
+
+        // Send the message and wait for response
+        // unsafe { host::api::message::send(self.process.id()) };
```

Comment:
I think this line was left behind by accident?

Reply:
Oh yes let me fix that